### PR TITLE
fix(search_family): Fix LOAD option behavior in the FT.AGGREGATE command

### LINKED ISSUE #3646: LOAD option is broken in the FT.AGGREGATE command
**Describe the bug**
Several snippets of the `FT.AGGREGATE` command do not work correct with the `LOAD` option:

Initialize:
```shell
127.0.0.1:6379> hset key:1 word item1 foo 10
(integer) 2
127.0.0.1:6379> hset key:2 word item2 foo 20
(integer) 2
127.0.0.1:6379> hset key:3 word item1 foo 30
(integer) 2
127.0.0.1:6379> ft.create index on hash schema word TAG foo NUMERIC
OK
```

1. LOAD should go before all other options, otherwise we should return an error:
```shell
127.0.0.1:6379> ft.aggregate index "*" GROUPBY 1 @word REDUCE SUM 1 @foo AS foo_total LOAD 1 foo_total
(error) LOAD cannot be applied after projectors or reducers
```

2. There may be several LOAD options at the beginning:
```shell
127.0.0.1:6379> ft.aggregate index "*" LOAD 1 @word LOAD 1 @foo
1) (integer) 1
2) 1) "word"
   2) "item1"
   3) "foo"
   4) "30"
3) 1) "word"
   2) "item2"
   3) "foo"
   4) "20"
4) 1) "word"
   2) "item1"
   3) "foo"
   4) "10"
```

3. Redis also supports the following command:
```shell
127.0.0.1:6379> ft.aggregate index "*" LOAD 1 @word GROUPBY 1 @word REDUCE SUM 1 @foo AS foo_total
1) (integer) 2
2) 1) "word"
   2) "item1"
   3) "foo_total"
   4) "40"
3) 1) "word"
   2) "item2"
   3) "foo_total"
   4) "20"
```
It still calculates the sum for the `@foo` field that has not been uploaded. ~~**But it seems to be a bug that does not match the documentation.**~~. Upd.: See #3659

### PR DESCRIPTION
**Pull request by BagritsevichStepan**
fixes #3646

### CODE REVIEW

**Comment by adiholden** [comment]
@BagritsevichStepan please add to description what was the bug and how it was fixed

**Comment by BagritsevichStepan** [comment]
This PR is a fork of [another PR](https://github.com/dragonflydb/dragonfly/pull/3657). 

This PR fixes bugs 1. and 2. from the issue #3646.

**Comment by chakaz** [comment]
Hey @BagritsevichStepan can you please ping me once you have merged the baseline PR and removed that diff from here? It's hard (and wasteful) to review forks IMHO, thanks!

**Comment by BagritsevichStepan** [comment]
> Hey @BagritsevichStepan can you please ping me once you have merged the baseline PR and removed that diff from here? It's hard (and wasteful) to review forks IMHO, thanks!

Yes, sure

File: src/server/search/search_family.cc [hunk 1]
```diff
@@ -252,19 +252,25 @@ optional<AggregateParams> ParseAggregatorParamsOrReply(CmdArgParser parser,
...
 
-  while (parser.HasNext()) {
-    // LOAD count field [field ...]
-    if (parser.Check("LOAD")) {
-      size_t num_fields = parser.Next<size_t>();
+  /*
+  Parse LOAD count field [field ...]
+
+  LOAD options are at the beginning of the query, so we need to parse them first
+  */
```
Comment (chakaz) [inline src/server/search/search_family.cc:259]:
```suggestion
  // Parse LOAD count field [field ...]
  // LOAD options are at the beginning of the query, so we need to parse them first
```
nit: It takes less vertical space like that :)

**Review by chakaz — DISMISSED** [review]

**Review by chakaz — APPROVED** [review]

**Review by chakaz — APPROVED** [review]

**Pull request merged**